src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_InterceptorTest.java

```diff
@@ -0,0 +1,191 @@
+package org.jboss.xavier.integrations.route;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Route;
+import org.apache.camel.component.rest.RestEndpoint;
+import org.apache.camel.impl.InterceptSendToEndpoint;
+import org.apache.camel.test.spring.CamelSpringBootRunner;
+import org.apache.camel.test.spring.MockEndpointsAndSkip;
+import org.apache.camel.test.spring.UseAdviceWith;
+import org.jboss.xavier.Application;
+import org.jboss.xavier.integrations.jpa.service.UserService;
+import org.jboss.xavier.integrations.util.TestUtil;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+
+@RunWith(CamelSpringBootRunner.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@MockEndpointsAndSkip("")
+@UseAdviceWith // Disables automatic start of Camel context
+@SpringBootTest(classes = {Application.class}, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+public class XmlRoutes_InterceptorTest {
+
+    @Autowired
+    CamelContext camelContext;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @SpyBean
+    private UserService userService;
+
+    @Value("${camel.component.servlet.mapping.context-path}")
+    String camel_context;
+
+    @Before
+    public void setup() {
+        camel_context = camel_context.substring(0, camel_context.indexOf("*"));
+    }
+
+    @Test
+    public void xmlRouteBuilder_AuthorizedInterceptor_GivenNoRHIdentity_ShouldReturnForbidden() throws Exception {
```

Comment:
It seems very similar to [`XmlRoutes_RestReportTest.xmlRouteBuilder_RestEndpoints_NoRHIdentityGiven_ShouldReturnForbidden`](https://github.com/project-xavier/xavier-integration/blob/master/src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java#L812) test: i think it worth having the interceptor test in this new `XmlRoutes_InterceptorTest` class so please move the checks from `xmlRouteBuilder_RestEndpoints_NoRHIdentityGiven_ShouldReturnForbidden` into this test/class.

Reply:
Since both tests were almost similar I've decided to merge them and move it to XmlRoutes_Interceptor.java Class